src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -117,12 +117,12 @@ public SocksBeginExBuilder typeId(
         public SocksBeginExBuilder address(
             String address) throws UnknownHostException
         {
-            if (vaildateIpv4(address))
+            if (IPV4_ADDRESS_MATCHER.get().reset(address).matches())
             {
                 final byte[] addressBytes = InetAddress.getByName(address).getAddress();
                 beginExRW.address(b -> b.ipv4Address(s -> s.set(addressBytes)));
```

Comment:
Now let's eliminate garbage from `IPv4` case.

Update the `Pattern` definition to capture each of 4 number parts of the `IPv4` address. This involves placing parentheses around the parts of the pattern that you wish to be able to extract after a successful pattern match.

So, `"192.168.0.1"` when applied to the pattern, the matcher will return `"192"` for group `1`, `"168"` for group `2`, etc.

```
final Matcher ipv4Matcher = IPV4_ADDRESS_MATCHER.get();
final byte[] addressBytes = new byte[4];
for (int i=0; i < addressBytes.length; i++)
{
    addressBytes[i] = Byte.parseByte(ipv4Matcher.group(i + 1));
}
beginExRW.address(b -> b.ipv4Address(s -> s.set(addressBytes)));
```
Now you need to figure out how to safely eliminate the `new byte[4]` above.

Reply:
Like writeBuffer or some buffers ??

Reply:
No, the answer you need is already demonstrated in this PR

Reply:
Tutorial on [Java Pattern capturing groups](https://docs.oracle.com/javase/tutorial/essential/regex/groups.html).